ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -0,0 +1,171 @@
+    zIndex: -1,
+    position: 'absolute'
+  },
+});
+
+const checkOpsBeforeUnload = (e) => {
+  if (Object.keys(compoundRuntimeOpStore.get()).length > 0) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+    e.preventDefault();
+    e.returnValue = '';
+  }
+};
+
+interface Props extends WithSpinnerOverlayProps {
+  onSignOut: () => {};
+  signOut: () => {};
+}
+
+export const SignedIn = (props: Props) => {
+  useEffect(() => props.hideSpinner(), []);
+
+  const [hideFooter, setHideFooter] = useState(null);
+  const [cdrVersionsInitialized, setCdrVersionsInitialized] = useState(false);
+  const [serverConfigInitialized, setServerConfigInitialized] = useState(false);
+  const [subscriptions, setSubscriptions] = useState([]);
+
+  const serverConfig = useStore(serverConfigStore);
+  const cdrVersions = useStore(cdrVersionStore);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', checkOpsBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', checkOpsBeforeUnload);
+    };
+  }, []);
+
+  useEffect(() => {
+    const subs = [];
+    // This handles detection of Angular-based routing data.
+    subs.push(routeConfigDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    // This handles detection of React-based routing data. During migrations,
+    // we assume React routing data will be set deeper/later in the component
+    // hierarchy, therefore it will generally take precedence over React.
+    subs.push(routeDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    setSubscriptions(subs);
+
+    return () => {
+      /*
+       * TODO RW-1920: we will only need one subscription after routeConfigDataStore and
+       *  routeDataStore are collapsed, and then we won't have to duck-type this
+       */
+      for (const s of subscriptions) {
+        s.unsubscribe();
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    /*
+     * TODO RW-6726 We want to block app rendering on the presence of server config data
+     *  so that route components don't try to lookup config data before it's available.
+     *  We should holistically figure out a Reacty way to do this.
+     *  See discussion on https://github.com/all-of-us/workbench/pull/4713
+     */
+    const checkStoresLoaded = async() => {
+      if (serverConfig.config) {
+        setServerConfigInitialized(true);
+        if (!profileStore.get().profile) {
+          await profileStore.get().load();
+        }
+        setInstitutionCategoryState(profileStore.get().profile.verifiedInstitutionalAffiliation);
+        if (hasRegisteredAccess(profileStore.get().profile.accessTierShortNames)) {
+          if (!cdrVersions) {
+            const cdrVersionsByTier = await cdrVersionsApi().getCdrVersionsByTier();
+            cdrVersionStore.set(cdrVersionsByTier);
+            setCdrVersionsInitialized(true);
```

Comment:
Likewise, I think it should be possible to directly derive this at the top of the function, or just reference it directly later on:

```
const serverConfigLoaded = !!serverConfig.config;
const cdrVersionsInitialized = !!cdrVersions;
```

or 

```
{serverConfig.config && cdrVersions && <...
```